### PR TITLE
Adds Mode chain

### PR DESCRIPTION
**Pull request by joshuanwankwo**
## PR-Codex overview
This PR adds a new `Mode` chain to the project.

### Detailed summary
- Added `Mode` chain to `chains/index.ts`
- Created `mode.ts` with chain definition for `Mode Mainnet`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: yarn.lock
Comment (jxom) [inline yarn.lock:1]: please revert

File: src/chains/index.ts [hunk 1]
```diff
@@ -203,6 +203,7 @@ export { zilliqaTestnet } from './definitions/zilliqaTestnet.js'
 export { zora } from './definitions/zora.js'
 export { zoraSepolia } from './definitions/zoraSepolia.js'
 export { zoraTestnet } from './definitions/zoraTestnet.js'
+export { mode } from './definitions/mode.js'
```
Comment (jxom) [inline src/chains/index.ts:206]: please arrange in alphabetical order

File: src/chains/definitions/mode.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  rpcUrls: {
+    default: {
+      http: ['https://mainnet.mode.network/'],
+    },
+  },
+  blockExplorers: {
+    default: {
+      name: 'Mode Explorer',
+      url: 'https://explorer.mode.network/',
+    //   apiUrl: '',
```
Comment (jxom) [inline src/chains/definitions/mode.ts:17]: remove

File: src/chains/definitions/mode.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
+  rpcUrls: {
+    default: {
+      http: ['https://mainnet.mode.network/'],
+    },
+  },
+  blockExplorers: {
+    default: {
+      name: 'Mode Explorer',
+      url: 'https://explorer.mode.network/',
```
Comment (jxom) [inline src/chains/definitions/mode.ts:16]:
```suggestion
      url: 'https://explorer.mode.network',
```

File: src/chains/definitions/mode.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { defineChain } from '../../utils/chain/defineChain.js'
+
+
+export const mode = /*#__PURE__*/ defineChain({
+  id: 34443,
+  name: 'Mode Mainnet',
+  nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
+  rpcUrls: {
+    default: {
+      http: ['https://mainnet.mode.network/'],
```
Comment (jxom) [inline src/chains/definitions/mode.ts:10]:
```suggestion
      http: ['https://mainnet.mode.network'],
```

File: src/chains/definitions/mode.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { defineChain } from '../../utils/chain/defineChain.js'
+
```
Comment (jxom) [inline src/chains/definitions/mode.ts:2]:
```suggestion
```

File: src/chains/index.ts (hunk 1 above)
Comment (joshuanwankwo) [reply src/chains/index.ts:206]: Fixed

File: src/chains/definitions/mode.ts (hunk 2 above)
Comment (joshuanwankwo) [reply src/chains/definitions/mode.ts:17]: Fixed

File: src/chains/definitions/mode.ts (hunk 3 above)
Comment (joshuanwankwo) [reply src/chains/definitions/mode.ts:16]: Fixed

File: src/chains/definitions/mode.ts (hunk 4 above)
Comment (joshuanwankwo) [reply src/chains/definitions/mode.ts:10]: Fixed

**Pull request merged**